packages/devtools_app/lib/src/memory/flutter/memory_graph_model.dart

```diff
@@ -0,0 +1,439 @@
+    });
+
+    // Prune libraries if all their classes are private or have zero instances.
+    groupByLibrary.removeWhere((libraryName, classes) {
+      classes.removeWhere((actual) =>
+          (controller.filterZeroInstances &&
+              actual.getInstances(this).isEmpty) ||
+          (controller.filterPrivateClasses && actual.name.startsWith('_')) ||
+          actual.name == internalClassName);
+
+      // Hide this library?
+      if (controller.libraryFilters.isLibraryFiltered(libraryName)) return true;
+
+      return controller.filterLibraryNoInstances && classes.isEmpty;
+    });
+
+    return true;
+  }
+
+  // TODO(terry): Need dominator graph for flow.
+  /// Compute all instances, needed for retained space.
+  void computeInstancesForClasses() {
+    if (!instancesComputed) {
+      for (var i = 0; i < elements.length; i++) {
+        final HeapGraphElementActual instance = elements[i];
+        instance.theClass.addInstance(instance);
+      }
+
+      instancesComputed = true;
+    }
+  }
+}
+
+abstract class HeapGraphElement {
+  /// Outbound references, i.e. this element points to elements in this list.
+  List<HeapGraphElement> _references;
+
+  void Function() referencesFiller;
+
+  List<HeapGraphElement> get references {
+    if (_references == null && referencesFiller != null) {
+      _references = [];
+      referencesFiller();
+    }
+    return _references;
+  }
+
+  String getPrettyPrint(Map<Uri, Map<String, List<String>>> prettyPrints) {
+    if (this is HeapGraphElementActual) {
+      final HeapGraphElementActual me = this;
+      if (me.theClass.toString() == '_OneByteString') {
+        return '"${me.origin.data}"';
+      }
+      if (me.theClass.toString() == '_SimpleUri') {
+        return '_SimpleUri['
+            "${me.getField("_uri").getPrettyPrint(prettyPrints)}]";
+      }
+      if (me.theClass.toString() == '_Uri') {
+        return "_Uri[${me.getField("scheme").getPrettyPrint(prettyPrints)}:"
+            "${me.getField("path").getPrettyPrint(prettyPrints)}]";
+      }
+      if (me.theClass is HeapGraphClassActual) {
+        final HeapGraphClassActual c = me.theClass;
+        final Map<String, List<String>> classToFields =
+            prettyPrints[c.libraryUri];
+        if (classToFields != null) {
+          final List<String> fields = classToFields[c.name];
+          if (fields != null) {
+            return '${c.name}[' +
+                fields.map((field) {
+                  return '$field: '
+                      '${me.getField(field)?.getPrettyPrint(prettyPrints)}';
+                }).join(', ') +
+                ']';
+          }
+        }
+      }
+    }
+    return toString();
+  }
+}
+
+/// Object marked for removal on next GC.
+class HeapGraphElementSentinel extends HeapGraphElement {
+  @override
+  String toString() => 'HeapGraphElementSentinel';
+}
+
+/// Live element.
+class HeapGraphElementActual extends HeapGraphElement {
```

Comment:
Maybe `HeapGraphElementLive` or `LiveHeapGraphElement` instead of `HeapGraphElementActual`?

Reply:
HeapGraphElementLive and HeapGraphClassLive